tests/src/test/scala/cats/tests/RegressionTests.scala

```diff
@@ -85,7 +85,7 @@ class RegressionTests extends CatsSuite {
 
   test("#500: foldMap - traverse consistency") {
     assert(
-      List(1,2,3).traverseU(i => Const(List(i))).getConst == List(1,2,3).foldMap(List(_))
+      List(1,2,3).traverse(i => Const.of[List[Int]](List(i))).getConst == List(1,2,3).foldMap(List(_))
```

Comment:
I think that specifying the type parameter here is an acceptable tradeoff for getting rid of the `Unapply` machinery, but paging @milessabin to see if he has any thoughts on this popping up.